x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/elasticsearch_monitor_states_adapter.ts

```diff
@@ -221,10 +227,28 @@ export class ElasticsearchMonitorStatesAdapter implements UMMonitorStatesAdapter
       monitors.push(...monitorBucketsToAdd);
       if (monitors.length > STATES.LEGACY_STATES_QUERY_SIZE) {
         monitors = monitors.slice(0, STATES.LEGACY_STATES_QUERY_SIZE);
+        isFinalPage = false;
```

Comment:
Will this work if we have a page size divisible by the number of monitors, or will we have an empty final page as the result?

Reply:
Err trying this out that seems to be the case, I think we need to search for one extra item than we want then trim that item off.

Reply:
👍